src/snapred/backend/recipe/algorithm/LoadGroupingDefinition.py

```diff
@@ -73,19 +77,25 @@ def validateInput(self) -> None:
         if file_extension in self.supported_calib_file_extensions:
             instrument_name = self.getProperty("InstrumentName").value
             instrument_file_name = self.getProperty("InstrumentFilename").value
-            absent_input_properties_count = sum(not prop for prop in [instrument_name, instrument_file_name])
-            if absent_input_properties_count == 0 or absent_input_properties_count == 2:
-                raise Exception("Either InstrumentName or InstrumentFilename must be specified, but not both.")
+            if instrument_name + instrument_file_name != "":
+                if not (instrument_name == "" or instrument_file_name == ""):
+                    raise Exception("Either InstrumentName or InstrumentFilename must be specified, but not both.")
+
```

Comment:
I think the original method of counting was cleaner than these string comparisons.  Why is an instrument donor not being considered a possibility for the case of H5/HD5/HDF files?

Reply:
The instrument donor is only used in the case of an XML file

Reply:
As far as I understand, all three methods can be used regardless of H5 or XML.  In the case of H5, the [LoadDiffCal](https://docs.mantidproject.org/nightly/algorithms/LoadDiffCal-v1.html) algorithm can be given an instrument donor through the `InputWorkspace` property, like so:
``` python
self.mantidSnapper.LoadDiffCal(
    "Loading grouping definition from calibration file...",
    Filename=grouping_file_name,
    InstrumentName=self.getProperty("InstrumentName").value,
    InstrumentFilename=self.getProperty("InstrumentFilename").value,
    InputWorkspace=self.getPropertyValue("InstrumentDonor"),
    MakeGroupingWorkspace=True,
    MakeCalWorkspace=False,
    MakeMaskWorkspace=False,
    WorkspaceName=output_ws_name,
)
```

Reply:
Ah ok. The call to LoadDiffcal below on line 96 was not making use of the InputWorkpsace property